primitives/src/lib.rs

```diff
@@ -87,7 +87,7 @@ pub type BlockId = generic::BlockId<Block>;
 pub use sp_runtime::OpaqueExtrinsic as UncheckedExtrinsic;
 
 #[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash))]
```

Comment:
Can you explain about the change here?

Reply:
The trait `std::hash::Hash` should be satisfied by CurrencyId enum when use it as the key of  `std::collections::HashMap`.